girder/models/file.py

```diff
@@ -93,23 +94,62 @@ def download(self, file, offset=0, headers=True, endByte=None,
         :type contentDisposition: str or None
         :type extraParameters: str or None
         """
+        if offset == 0:
+            events.trigger('model.file.download.before', info={'file': file})
+        else:
+            events.trigger('model.file.download.during', info={'file': file})
```

Comment:
I would combine "before" and "during" into one called "model.file.download.request" which has startByte and endByte keys, and change the other one to be called "model.file.download.complete", to be more expressive of the fact that the file was assumed to be completely downloaded.